src/DeepLinkModule/javasource/deeplink/actions/StartDeeplinkJava.java

```diff
@@ -93,87 +93,92 @@ public String toString()
 		private String loginLocation = emptyStringToNull((String) Core.getConfiguration().getConstantValue("DeepLink.LoginLocation"));
 		private static final String	INDEX_CONSTANT	= "DeepLink.IndexPage";
 		private static final String	DEFAULT_INDEX_LOCATION	= "index.html";
-		
+
 		@Override
 		public void processRequest(IMxRuntimeRequest request,
-				IMxRuntimeResponse response, String arg2) throws Exception 
+				IMxRuntimeResponse response, String arg2) throws Exception
 		{
 		    /** BJHL - CC0000000100277
-             * Office (esp. Word) opens links by fetching the URL using its own cookies, following 
+             * Office (esp. Word) opens links by fetching the URL using its own cookies, following
              * any redirects and opening the resulting location in a browser. The browser will then
              * send its cookies, which ultimately results in a session mismatch between word and
              * the browser, causing the pending deeplink to be created for the wrong session. This
-             * is a know problem with word: 
+             * is a know problem with word:
              *   https://stackoverflow.com/questions/1421608/how-does-ms-word-open-a-hyperlink
              * To "fix" this, ignore any requests where the user-agent contains "office". Sending
-             * a 200 OK tells word the end of the redirect is reached, and it will happily open the 
-             * original URL in your default browser... 
+             * a 200 OK tells word the end of the redirect is reached, and it will happily open the
+             * original URL in your default browser...
              */
             if (request.getHeader("User-Agent").toLowerCase().contains("office")) {
                 response.setStatus(IMxRuntimeResponse.OK);
                 return;
             }
-            
+
 			try {
+				String requestPath = "";
 				String[] args = request.getResourcePath().split("/");
-				for (int j1 = 0; j1 < args.length; j1++) //args will be used in queries, so escape
+				for (int j1 = 0; j1 < args.length; j1++) {//args will be used in queries, so escape
 					args[j1] = StringEscapeUtils.escapeXml(args[j1]);
-				
+					if (j1 > 1) {
+						requestPath = (requestPath == "" ? "" : requestPath + "/") + args[j1];

```

Comment:
RequestPath starts off as "", and if empty string, it sets it to empty string. So it never gets any other value here!